src/sortingtraining/SortTester.java

```diff
@@ -11,43 +11,69 @@
 
 public class SortTester {
 
+	private Strategy strategy;
+
 	public static void main(String[] args) {
 		new SortTester().run();
 	}
 
-
 	void run() {
 
 		try (Scanner in = new Scanner(System.in)) {
 
-			System.out.print("Type size of your Array with random numbers: ");
+			int[] numbers;
 
-			final int[] numbers = ArrayGeneratorUtil.generate(Integer.parseInt(in.nextLine()));
+			numbers = getInts(in);
 
-			System.out.println("Your new Array: " + Arrays.toString(numbers));
+			mainMenu(in, numbers);
+		}
+	}
 
+	private void mainMenu(Scanner in, int[] numbers) {
 
-			while (true) {
+		while (true) {
 
-				System.out.println("");
-				System.out.print("Type a sorting strategy (BUBBLE, BUCKET, INSERTION, QUICK, DEFAULT, ALL(compare all), EXIT): ");
-				String strategy = in.nextLine();
+			while (true) {
+				System.out.print(
+						"Type a sorting strategy (BUBBLE, BUCKET, INSERTION, QUICK, DEFAULT, ALL(compare all), EXIT): ");
+				try {
+					strategy = Strategy.valueOf(in.nextLine().toUpperCase());
 
-				if (strategy.equals("EXIT")) {
 					break;
+				} catch (IllegalArgumentException e) {
+					System.out.println("Wrong input, try again!");
+				}
+			}
 
-				} else if (strategy.equals("ALL")) {
-					ComparingUtil comparingUtil = new ComparingUtil();
-					comparingUtil.compare(numbers);
-
-				} else {
+			if (strategy == Strategy.EXIT) {
+				break;
 
-					SortingUtil sortingUtil = new SortingUtil(new SortingStrategySwitchResolver());
-					sortingUtil.sort(numbers, strategy);
+			} else if (strategy == Strategy.ALL) {
+				ComparingUtil comparingUtil = new ComparingUtil();
+				comparingUtil.compare(numbers);
 
-				}
+			} else {
+				SortingUtil sortingUtil = new SortingUtil(new SortingStrategySwitchResolver());
+				sortingUtil.sort(numbers, strategy);
 			}
 		}
 	}
 
+	private int[] getInts(Scanner in) {
+		int[] numbers;
+		while (true) {
+			System.out.print("Type size of your Array with random numbers: ");
```

Comment:
How about more User Friendly approach: first collect from user all required data, and validate it, than pick randoms  and than finally do the computation ? :)

Please remember about new line feed: \n